File: gitdot-api-ts/src/endpoint/runner/list-runners.ts
import { z } from "zod";
import { RunnerResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListRunnersRequest = z.object({});
export type ListRunnersRequest = z.infer<typeof ListRunnersRequest>;

export const ListRunnersResponse = z.array(RunnerResource);
export type ListRunnersResponse = z.infer<typeof ListRunnersResponse>;

export const ListRunners = {
  path: "/ci/runner/{owner}",
  method: "GET",
  request: ListRunnersRequest,
  response: ListRunnersResponse,
} as const satisfies Endpoint;
export type ListRunners = typeof ListRunners;

File: gitdot-api/src/endpoint/runner/list_runners.rs
use crate::{endpoint::Endpoint, resource::RunnerResource};

pub struct ListRunners;

impl Endpoint for ListRunners {
    const PATH: &'static str = "/ci/runner/{owner}";
    const METHOD: http::Method = http::Method::GET;

    type Request = ();
    type Response = Vec<RunnerResource>;
}

pub type ListRunnersRequest = ();
pub type ListRunnersResponse = Vec<RunnerResource>;
